internal/mode/static/state/dataplane/configuration_test.go

```diff
@@ -3339,3 +3340,138 @@ func TestBuildAdditions(t *testing.T) {
 		})
 	}
 }
+
+func TestCreatePassthroughServers(t *testing.T) {
+	testGraph := graph.Graph{
+		Gateway: &graph.Gateway{
+			Listeners: make([]*graph.Listener, 0, 1),
+		},
+	}
+	l4routes := make(map[graph.L4RouteKey]*graph.L4Route)
+
+	nsname := types.NamespacedName{
+		Namespace: "default",
+		Name:      "secure-app",
+	}
+
+	key := graph.L4RouteKey{NamespacedName: nsname}
+	l4routes[key] = &graph.L4Route{
+		Spec: graph.L4RouteSpec{
+			Hostnames: []v1.Hostname{"app.example.com", "cafe.example.com"},
+			BackendRef: graph.BackendRef{
+				SvcNsName: nsname,
+				ServicePort: apiv1.ServicePort{
+					Name:        "https",
+					Protocol:    "TCP",
+					AppProtocol: nil,
+					Port:        8443,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8443,
+					},
+				},
+				Valid: true,
+			},
+		},
+		Valid: true,
+	}
+	testGraph.Gateway.Listeners = append(testGraph.Gateway.Listeners, &graph.Listener{
```

Comment:
I recommend defining this listener inline when you create the test graph. Same with the L4 routes